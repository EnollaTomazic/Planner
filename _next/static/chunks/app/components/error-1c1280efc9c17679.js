(self.webpackChunk_N_E=self.webpackChunk_N_E||[]).push([[872],{50960:(e,s,r)=>{"use strict";r.d(s,{A:()=>o,R:()=>c});var a=r(95155),n=r(12115),i=r(52619),t=r.n(i),l=r(24625);function c(e){let{error:s,reset:r,title:i,description:c,retryLabel:o="Try again",homeLabel:d="Go to dashboard",homeHref:h="/"}=e;return(0,n.useEffect)(()=>{console.error(s)},[s]),(0,a.jsx)(l.qK,{as:"section",role:"alert","aria-live":"assertive",className:"py-[var(--space-8)]",children:(0,a.jsxs)("div",{className:"flex max-w-prose flex-col gap-[var(--space-4)]",children:[(0,a.jsxs)("div",{className:"space-y-[var(--space-2)]",children:[(0,a.jsx)("h1",{className:"text-display-sm font-semibold text-foreground",children:i}),(0,a.jsx)("p",{className:"text-body text-muted-foreground",children:c})]}),(0,a.jsxs)("div",{className:"flex flex-wrap gap-[var(--space-3)]",children:[(0,a.jsx)(l.$n,{variant:"primary",onClick:r,children:o}),(0,a.jsx)(l.$n,{asChild:!0,variant:"ghost",children:(0,a.jsx)(t(),{href:h,children:d})})]})]})})}function o(e){return(0,a.jsx)(c,{...e,title:"Something went wrong",description:"This section hit an error, but the rest of Planner is still running. Try again or return home."})}},59729:(e,s,r)=>{"use strict";r.r(s),r.d(s,{default:()=>a.A});var a=r(50960)},94822:(e,s,r)=>{Promise.resolve().then(r.bind(r,59729))}},e=>{e.O(0,[6428,9295,2619,4625,8441,1255,7358],()=>e(e.s=94822)),_N_E=e.O()}]);