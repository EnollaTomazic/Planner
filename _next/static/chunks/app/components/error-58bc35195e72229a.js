(self.webpackChunk_N_E=self.webpackChunk_N_E||[]).push([[872],{28558:(e,t,r)=>{"use strict";r.d(t,{D:()=>a});let n=(0,r(10161).h)("clipboard");async function a(e){if("undefined"==typeof navigator||"undefined"==typeof document)return;let t=navigator.clipboard,r=null;if(null==t?void 0:t.writeText)try{await t.writeText(e);return}catch(e){r=e}if((null==t?void 0:t.write)&&"undefined"!=typeof ClipboardItem)try{let r=new ClipboardItem({"text/plain":new Blob([e],{type:"text/plain"})});await t.write([r]);return}catch(e){r=e}if(!document.body)return void n.warn("Unable to copy text because document.body is not available.",r);let a=document.createElement("textarea");a.value=e,a.classList.add("clipboard-copy-buffer"),document.body.appendChild(a),a.select();let l=!1;try{if("function"==typeof document.execCommand&&(l=document.execCommand("copy")))return}catch(e){r=e}finally{var i;null==(i=document.getSelection())||i.removeAllRanges(),a.remove()}l||n.warn("Failed to copy text with the available clipboard strategies.",r)}},34837:(e,t,r)=>{"use strict";r.d(t,{x:()=>T});var n,a,l,i,o,c,s,d=r(10161),u=r(87358);let p=d.UG.child("sentry"),m=p.child("capture"),f=(null==(n=u.env.NEXT_PUBLIC_SENTRY_DSN)?void 0:n.trim())||(null==(a=u.env.SENTRY_DSN)?void 0:a.trim())||"",y=(null==(l=u.env.NEXT_PUBLIC_SENTRY_ENVIRONMENT)?void 0:l.trim())||(null==(i=u.env.SENTRY_ENVIRONMENT)?void 0:i.trim())||"production",v=(null==(o=u.env.NEXT_PUBLIC_SENTRY_TRACES_SAMPLE_RATE)?void 0:o.trim())||(null==(c=u.env.SENTRY_TRACES_SAMPLE_RATE)?void 0:c.trim())||"",h=!!f,x=!1,b=null,N=null,g=null!=(s=function(e){if(!e)return;let t=Number.parseFloat(e);return Number.isNaN(t)?void p.warn("Invalid Sentry sample rate provided",{value:e}):Math.min(Math.max(t,0),1)}(v))?s:.1;async function E(){return h?N||(b||(b=Promise.all([r.e(4850),r.e(7141)]).then(r.bind(r,17141)).then(e=>{if(N=e,!x)try{e.init({dsn:f,environment:y,enabled:h,tracesSampleRate:g,replaysSessionSampleRate:0,replaysOnErrorSampleRate:0,profilesSampleRate:0}),x=!0}catch(e){return p.error("Failed to initialize Sentry",e),null}return N}).catch(e=>(p.error("Failed to load Sentry runtime",e),N=null,null)).finally(()=>{b=null})),b):null}async function w(e,t){if(!h)return;let r=await E();if(!r)return void p.warn("captureException skipped because Sentry is unavailable",e);try{r.captureException(e,t)}catch(t){m.error("Failed to capture exception with Sentry",t,e)}}let S=d.UG.child("error-boundary");function T(e){let{boundary:t,error:r,info:n,tags:a,extra:l}=e;S.error("Error boundary captured an exception",{boundary:t,error:r,info:n,tags:a,extra:l});let i={boundary:t,...null!=a?a:{}},o={boundary:t,...null!=l?l:{}};n&&(o.info=n),w(r,{tags:i,extra:o})}},50960:(e,t,r)=>{"use strict";r.d(t,{A:()=>f,R:()=>m});var n=r(95155),a=r(12115),l=r(52619),i=r.n(l),o=r(91182),c=r(10161),s=r(34837),d=r(28558),u=r(27566);let p=(0,c.h)("route:error-boundary");function m(e){let{error:t,reset:r,title:l,description:c,retryLabel:m="Try again",homeLabel:f="Go to dashboard",homeHref:y=(0,u.hd)("/")}=e,v=(0,a.useMemo)(()=>(null==t?void 0:t.digest)?t.digest:"undefined"!=typeof crypto&&"function"==typeof crypto.randomUUID?crypto.randomUUID():"fallback-".concat(Date.now().toString(36),"-").concat(Math.random().toString(36).slice(2,8)),[t]),h=(0,a.useMemo)(()=>{var e;let r="string"==typeof(null==t?void 0:t.stack)?t.stack:"",n=null!=(e=null==t?void 0:t.message)?e:"Unknown error",a=r.includes(n)?r:[n,r].filter(Boolean).join("\n"),l=(null==t?void 0:t.digest)?"Digest: ".concat(t.digest):"";return["Planner error ".concat(v),l,a].filter(Boolean).join("\n\n")},[t,v]),[x,b]=(0,a.useState)("");(0,a.useEffect)(()=>{(0,s.x)({boundary:"route",error:t,extra:{digest:null==t?void 0:t.digest,errorId:v},tags:{errorId:v}})},[t,v]),(0,a.useEffect)(()=>{if(!x)return;let e=window.setTimeout(()=>b(""),4e3);return()=>window.clearTimeout(e)},[x]);let N=(0,a.useCallback)(async()=>{try{await (0,d.D)(h),b("Stack copied to clipboard.")}catch(e){p.warn("Failed to copy stack trace from route error boundary",e),b("We couldn't copy the stack. Select and copy it manually if needed.")}},[h]);return(0,n.jsx)(o.qK,{as:"section",role:"alert","aria-live":"assertive",className:"py-[var(--space-8)]",children:(0,n.jsxs)("div",{className:"flex max-w-prose flex-col gap-[var(--space-4)]",children:[(0,n.jsxs)("div",{className:"space-y-[var(--space-2)]",children:[(0,n.jsx)("h1",{className:"text-title-lg font-semibold text-foreground",children:l}),(0,n.jsx)("p",{className:"text-body text-muted-foreground",children:c})]}),(0,n.jsxs)("div",{className:"flex flex-wrap gap-[var(--space-3)]",children:[(0,n.jsx)(o.$n,{variant:"default",onClick:r,children:m}),(0,n.jsx)(o.$n,{asChild:!0,variant:"quiet",children:(0,n.jsx)(i(),{href:(0,u.hd)(y),children:f})}),(0,n.jsx)(o.$n,{type:"button",variant:"quiet",size:"sm",onClick:N,children:"Copy stack"})]}),(0,n.jsxs)("div",{className:"space-y-[var(--space-1)]",children:[(0,n.jsxs)("p",{className:"text-label text-muted-foreground",children:["Reference ID: ",(0,n.jsx)("span",{className:"font-mono text-foreground",children:v})]}),(0,n.jsx)("div",{className:"sr-only","aria-live":"polite",children:x}),x?(0,n.jsx)("p",{className:"text-label text-muted-foreground",children:x}):null]})]})})}function f(e){return(0,n.jsx)(m,{...e,title:"Something went wrong",description:"This section hit an error, but the rest of Planner is still running. Try again or return home."})}},59729:(e,t,r)=>{"use strict";r.r(t),r.d(t,{default:()=>n.A});var n=r(50960)},94822:(e,t,r)=>{Promise.resolve().then(r.bind(r,59729))}},e=>{e.O(0,[1925,603,8087,3498,2619,1182,8441,1255,7358],()=>e(e.s=94822)),_N_E=e.O()}]);
//# sourceMappingURL=error-58bc35195e72229a.js.map