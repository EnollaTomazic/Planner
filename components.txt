1:"$Sreact.fragment"
2:I[41402,["678","static/chunks/678-3b8ca7f6c2bfafc6.js","619","static/chunks/619-f072ac750404f9da.js","625","static/chunks/625-de68f502e68d6418.js","177","static/chunks/app/layout-ff43e66cdbce5daf.js"],""]
3:I[28727,["678","static/chunks/678-3b8ca7f6c2bfafc6.js","619","static/chunks/619-f072ac750404f9da.js","625","static/chunks/625-de68f502e68d6418.js","177","static/chunks/app/layout-ff43e66cdbce5daf.js"],"default"]
4:I[98002,["678","static/chunks/678-3b8ca7f6c2bfafc6.js","619","static/chunks/619-f072ac750404f9da.js","625","static/chunks/625-de68f502e68d6418.js","177","static/chunks/app/layout-ff43e66cdbce5daf.js"],"default"]
5:I[9766,[],""]
6:I[98924,[],""]
7:I[41061,["678","static/chunks/678-3b8ca7f6c2bfafc6.js","619","static/chunks/619-f072ac750404f9da.js","689","static/chunks/689-e5d24021d927ff8f.js","736","static/chunks/736-4db4399c886d749f.js","634","static/chunks/634-bfd3c8aa4039d99b.js","625","static/chunks/625-de68f502e68d6418.js","373","static/chunks/373-f15cb54580f283eb.js","108","static/chunks/108-03d12cb04b17ced4.js","522","static/chunks/522-4c109c4a359a5202.js","879","static/chunks/879-fd6a10fb62d46999.js","423","static/chunks/app/components/page-73e48587b2bb0ffd.js"],"default"]
8:I[52619,["678","static/chunks/678-3b8ca7f6c2bfafc6.js","619","static/chunks/619-f072ac750404f9da.js","625","static/chunks/625-de68f502e68d6418.js","345","static/chunks/app/not-found-e4eee859b0a932fb.js"],""]
9:I[93462,["678","static/chunks/678-3b8ca7f6c2bfafc6.js","619","static/chunks/619-f072ac750404f9da.js","689","static/chunks/689-e5d24021d927ff8f.js","736","static/chunks/736-4db4399c886d749f.js","634","static/chunks/634-bfd3c8aa4039d99b.js","625","static/chunks/625-de68f502e68d6418.js","373","static/chunks/373-f15cb54580f283eb.js","108","static/chunks/108-03d12cb04b17ced4.js","522","static/chunks/522-4c109c4a359a5202.js","879","static/chunks/879-fd6a10fb62d46999.js","423","static/chunks/app/components/page-73e48587b2bb0ffd.js"],"default"]
f:I[57150,[],""]
:HL["/Planner/_next/static/css/eee3817ff2c519e0.css","style"]
:HL["/Planner/_next/static/css/cb150f8d5a7ebde5.css","style"]
:HL["/Planner/_next/static/css/9d79e8c8cefd0d15.css","style"]
:HL["/Planner/_next/static/css/4e282041685a4199.css","style"]
:HL["/Planner/_next/static/css/fac86c4fdf6106a7.css","style"]
0:{"P":null,"b":"Fmq1Vd77VM9lxh_JyjaIU","p":"/Planner","c":["","components"],"i":false,"f":[[["",{"children":["components",{"children":["__PAGE__",{}]}]},"$undefined","$undefined",true],["",["$","$1","c",{"children":[[["$","link","0",{"rel":"stylesheet","href":"/Planner/_next/static/css/eee3817ff2c519e0.css","precedence":"next","crossOrigin":"$undefined","nonce":"$undefined"}],["$","link","1",{"rel":"stylesheet","href":"/Planner/_next/static/css/cb150f8d5a7ebde5.css","precedence":"next","crossOrigin":"$undefined","nonce":"$undefined"}],["$","link","2",{"rel":"stylesheet","href":"/Planner/_next/static/css/9d79e8c8cefd0d15.css","precedence":"next","crossOrigin":"$undefined","nonce":"$undefined"}]],["$","html",null,{"lang":"en","className":"theme-lg","suppressHydrationWarning":true,"style":{"--asset-noise-url":"url('/Planner/noise.svg')","--asset-glitch-gif-url":"url('/Planner/glitch-gif.gif')"},"children":[["$","head",null,{"children":["$","$L2",null,{"id":"theme-bootstrap","strategy":"beforeInteractive","src":"/Planner/scripts/theme-bootstrap.js"}]}],["$","body",null,{"className":"min-h-screen bg-background text-foreground glitch-root","children":[["$","a",null,{"className":"fixed left-[var(--space-4)] top-[var(--space-4)] z-50 inline-flex items-center rounded-[var(--radius-lg)] bg-background px-[var(--space-4)] py-[var(--space-2)] text-ui font-medium text-foreground shadow-outline-subtle outline-none transition-all duration-[var(--dur-quick)] ease-[var(--ease-out)] opacity-0 -translate-y-full pointer-events-none focus-visible:translate-y-0 focus-visible:opacity-100 focus-visible:pointer-events-auto focus-visible:shadow-ring focus-visible:no-underline focus-visible:outline-none hover:shadow-ring focus-visible:active:translate-y-[var(--space-1)]","href":"#main-content","children":"Skip to main content"}],["$","$L3",null,{"children":[["$","div",null,{"aria-hidden":true,"className":"page-backdrop","children":["$","div",null,{"className":"page-shell","children":["$","div",null,{"className":"page-backdrop__layer"}]}]}],["$","$L4",null,{}],["$","div",null,{"className":"fixed bottom-[var(--space-4)] left-[var(--space-4)] z-50 pointer-events-none select-none","children":["$","svg",null,{"className":"w-[var(--space-8)] h-[var(--space-8)] text-accent motion-safe:animate-[cat-float_3s_ease-in-out_infinite]","role":"img","aria-label":"Cat companion","tabIndex":-1,"focusable":"false","viewBox":"0 0 24 24","fill":"currentColor","children":["$","path",null,{"d":"M12 5c.67 0 1.35.09 2 .26 1.78-2 5.03-2.84 6.42-2.26 1.4.58-.42 7-.42 7 .57 1.07 1 2.24 1 3.44C21 17.9 16.97 21 12 21s-9-3-9-7.56c0-1.25.5-2.4 1-3.44 0 0-1.89-6.42-.5-7 1.39-.58 4.72.23 6.5 2.23A9.04 9.04 0 0 1 12 5Z"}]}]}],["$","div",null,{"id":"main-content","tabIndex":-1,"className":"relative z-10","children":["$","$L5",null,{"parallelRouterKey":"children","error":"$undefined","errorStyles":"$undefined","errorScripts":"$undefined","template":["$","$L6",null,{}],"templateStyles":"$undefined","templateScripts":"$undefined","notFound":[["$","main",null,{"className":"flex min-h-screen flex-col items-center justify-center gap-4 p-6 text-center","aria-labelledby":"not-found-header","children":["$","$L7",null,{"className":"rounded-card r-card-lg px-4 py-4","header":{"id":"not-found-header","heading":"Page not found","icon":["$","svg",null,{"ref":"$undefined","xmlns":"http://www.w3.org/2000/svg","width":24,"height":24,"viewBox":"0 0 24 24","fill":"none","stroke":"currentColor","strokeWidth":2,"strokeLinecap":"round","strokeLinejoin":"round","className":"lucide lucide-circle-alert opacity-80","aria-hidden":"true","children":[["$","circle","1mglay",{"cx":"12","cy":"12","r":"10"}],["$","line","1pkeuh",{"x1":"12","x2":"12","y1":"8","y2":"12"}],["$","line","4dfq90",{"x1":"12","x2":"12.01","y1":"16","y2":"16"}],"$undefined"]}]},"hero":{"frame":false,"heading":"This page does not exist","actions":["$","$L8",null,{"href":"/","children":["$","$L9",null,{"className":"px-4","children":"Go home"}]}],"children":["$","p",null,{"className":"text-ui text-muted-foreground","children":"The page you are looking for does not exist."}]}}]}],[]],"forbidden":"$undefined","unauthorized":"$undefined"}]}]]}]]}]]}]]}],{"children":["components",["$","$1","c",{"children":[null,["$","$L5",null,{"parallelRouterKey":"children","error":"$undefined","errorStyles":"$undefined","errorScripts":"$undefined","template":["$","$L6",null,{}],"templateStyles":"$undefined","templateScripts":"$undefined","notFound":"$undefined","forbidden":"$undefined","unauthorized":"$undefined"}]]}],{"children":["__PAGE__",["$","$1","c",{"children":["$La",["$Lb","$Lc"],"$Ld"]}],{},null,false]},null,false]},null,false],"$Le",false]],"m":"$undefined","G":["$f",[]],"s":false,"S":true}
10:"$Sreact.suspense"
11:I[33173,["678","static/chunks/678-3b8ca7f6c2bfafc6.js","619","static/chunks/619-f072ac750404f9da.js","689","static/chunks/689-e5d24021d927ff8f.js","736","static/chunks/736-4db4399c886d749f.js","634","static/chunks/634-bfd3c8aa4039d99b.js","625","static/chunks/625-de68f502e68d6418.js","373","static/chunks/373-f15cb54580f283eb.js","108","static/chunks/108-03d12cb04b17ced4.js","522","static/chunks/522-4c109c4a359a5202.js","879","static/chunks/879-fd6a10fb62d46999.js","423","static/chunks/app/components/page-73e48587b2bb0ffd.js"],"default"]
12:I[26628,["678","static/chunks/678-3b8ca7f6c2bfafc6.js","619","static/chunks/619-f072ac750404f9da.js","689","static/chunks/689-e5d24021d927ff8f.js","736","static/chunks/736-4db4399c886d749f.js","634","static/chunks/634-bfd3c8aa4039d99b.js","625","static/chunks/625-de68f502e68d6418.js","373","static/chunks/373-f15cb54580f283eb.js","108","static/chunks/108-03d12cb04b17ced4.js","522","static/chunks/522-4c109c4a359a5202.js","879","static/chunks/879-fd6a10fb62d46999.js","423","static/chunks/app/components/page-73e48587b2bb0ffd.js"],"default"]
13:I[24431,[],"OutletBoundary"]
15:I[15278,[],"AsyncMetadataOutlet"]
17:I[24431,[],"ViewportBoundary"]
19:I[24431,[],"MetadataBoundary"]
a:["$","$10",null,{"fallback":["$","div",null,{"className":"flex justify-center p-6","children":["$","$L11",null,{}]}],"children":["$","$L12",null,{}]}]
b:["$","link","0",{"rel":"stylesheet","href":"/Planner/_next/static/css/4e282041685a4199.css","precedence":"next","crossOrigin":"$undefined","nonce":"$undefined"}]
c:["$","link","1",{"rel":"stylesheet","href":"/Planner/_next/static/css/fac86c4fdf6106a7.css","precedence":"next","crossOrigin":"$undefined","nonce":"$undefined"}]
d:["$","$L13",null,{"children":["$L14",["$","$L15",null,{"promise":"$@16"}]]}]
e:["$","$1","h",{"children":[null,[["$","$L17",null,{"children":"$L18"}],null],["$","$L19",null,{"children":["$","div",null,{"hidden":true,"children":["$","$10",null,{"fallback":null,"children":"$L1a"}]}]}]]}]
18:[["$","meta","0",{"charSet":"utf-8"}],["$","meta","1",{"name":"viewport","content":"width=device-width, initial-scale=1"}]]
14:null
16:{"metadata":[["$","title","0",{"children":"Components Â· Planner"}],["$","meta","1",{"name":"description","content":"Browse Planner UI building blocks and examples."}]],"error":null,"digest":"$undefined"}
1a:"$16:metadata"
