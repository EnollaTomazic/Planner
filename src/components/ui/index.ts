// Auto-generated by scripts/generate-ui-index.ts
// Do not edit directly.
export { default as AnimationToggle } from "./AnimationToggle";
export { default as Badge } from "./Badge";
export * from "./Badge";
export { default as Label } from "./Label";
export * from "./Label";
export { default as Select } from "./Select";
export * from "./Select";
export { default as Progress } from "./feedback/Progress";
export { default as Snackbar } from "./feedback/Snackbar";
export { default as Spinner } from "./feedback/Spinner";
export { default as Header } from "./layout/Header";
export * from "./layout/Header";
export { default as Hero2 } from "./layout/Hero2";
export * from "./layout/Hero2";
export { default as SectionCard } from "./layout/SectionCard";
export { default as Split } from "./layout/Split";
export { default as TabBar } from "./layout/TabBar";
export * from "./layout/TabBar";
export { default as TitleBar } from "./layout/TitleBar";
export { default as SideSelector } from "./league/SideSelector";
export * from "./league/SideSelector";
export { default as PillarBadge } from "./league/pillars/PillarBadge";
export { default as PillarSelector } from "./league/pillars/PillarSelector";
export { default as Button } from "./primitives/Button";
export * from "./primitives/Button";
export { default as Card } from "./primitives/Card";
export * from "./primitives/Card";
export { default as FieldShell } from "./primitives/FieldShell";
export * from "./primitives/FieldShell";
export * from "./primitives/GlitchSegmented";
export { default as IconButton } from "./primitives/IconButton";
export * from "./primitives/IconButton";
export { default as Input } from "./primitives/Input";
export * from "./primitives/Input";
export * from "./primitives/Neu";
export { default as SearchBar } from "./primitives/SearchBar";
export * from "./primitives/SearchBar";
export { default as Textarea } from "./primitives/Textarea";
export * from "./primitives/Textarea";
export { default as AnimatedSelect } from "./selects/AnimatedSelect";
export * from "./selects/AnimatedSelect";
export { default as ThemeToggle } from "./theme/ThemeToggle";
export { default as CheckCircle } from "./toggles/CheckCircle";
export * from "./toggles/NeonIcon";
export { default as Toggle } from "./toggles/toggle";
