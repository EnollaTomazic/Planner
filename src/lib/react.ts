import * as React from "react";

export const hasTextContent = (node: React.ReactNode): boolean => {
  if (node === null || node === undefined) return false;
  if (typeof node === "boolean") return false;
  if (typeof node === "string") return node.trim().length > 0;
  if (typeof node === "number") return true;
  if (Array.isArray(node)) return node.some((item) => hasTextContent(item));
  if (React.isValidElement<{ children?: React.ReactNode }>(node)) {
    return hasTextContent(node.props.children ?? null);
  }
  return false;
};

export function observeMediaQuery(
  query: string,
  onChange: (matches: boolean, mediaQuery: MediaQueryList) => void,
): (() => void) | undefined {
  if (typeof window === "undefined" || typeof window.matchMedia !== "function") {
    return undefined;
  }

  let mediaQuery: MediaQueryList;

  try {
    mediaQuery = window.matchMedia(query);
  } catch {
    return undefined;
  }

  const handleChange = (event: MediaQueryListEvent): void => {
    onChange(event.matches, mediaQuery);
  };

  const handleLegacyChange = (): void => {
    onChange(mediaQuery.matches, mediaQuery);
  };

  onChange(mediaQuery.matches, mediaQuery);

  if (typeof mediaQuery.addEventListener === "function") {
    mediaQuery.addEventListener("change", handleChange);
    return () => {
      mediaQuery.removeEventListener("change", handleChange);
    };
  }

  if (typeof mediaQuery.addListener === "function") {
    mediaQuery.addListener(handleLegacyChange);
    return () => {
      mediaQuery.removeListener(handleLegacyChange);
    };
  }

  return undefined;
}

export function useMatchMedia(
  query: string,
  getServerSnapshot?: () => boolean,
): boolean {
  const getMatches = React.useCallback(() => {
    if (typeof window === "undefined") {
      return typeof getServerSnapshot === "function" ? getServerSnapshot() : false;
    }

    if (typeof window.matchMedia !== "function") {
      return false;
    }

    try {
      return window.matchMedia(query).matches;
    } catch {
      return false;
    }
  }, [query, getServerSnapshot]);

  const [matches, setMatches] = React.useState<boolean>(getMatches);

  React.useEffect(() => {
    return observeMediaQuery(query, (nextMatches) => {
      setMatches(nextMatches);
    });
  }, [query]);

  return matches;
}
