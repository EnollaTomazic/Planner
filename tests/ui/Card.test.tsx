import * as React from "react";
import { cleanup, render } from "@testing-library/react";
import { afterEach, describe, expect, it } from "vitest";

import { Card } from "@/components/ui";
import { readFileSync } from "node:fs";
import path from "node:path";
import cardStyles from "@/components/ui/primitives/Card.module.css";

afterEach(cleanup);

describe("Card", () => {
  const cardCssSource = readFileSync(
    path.resolve(
      process.cwd(),
      "src/components/ui/primitives/Card.module.css",
    ),
    "utf8",
  );

  it("maps the glitch overlay opacity to the card token", () => {
    const root = document.documentElement;
    const previous = root.style.getPropertyValue("--glitch-overlay-opacity-card");
    const hadPrevious = previous.trim().length > 0;
    root.style.setProperty("--glitch-overlay-opacity-card", "0.64");

    try {
      const { container } = render(<Card glitch data-text="glitch" />);
      const element = container.firstElementChild as HTMLElement | null;

      expect(element).not.toBeNull();
      if (!element) {
        throw new Error("Card root element was not rendered");
      }

      const glitchClass = cardStyles.glitch;
      expect(glitchClass).toBeDefined();
      expect(element.classList.contains(glitchClass)).toBe(true);

      const cssMatch = cardCssSource.match(
        /\.glitch\s*\{[^}]*--glitch-overlay-opacity:[^;}]+;[^}]*\}/,
      );
      if (!cssMatch) {
        throw new Error("Glitch overlay declaration was not found in CSS");
      }

      const computedToken = getComputedStyle(element)
        .getPropertyValue("--glitch-overlay-opacity-card")
        .trim();

      const snippet = cssMatch[0].replace(/\s+/g, " ").trim();

      expect({ computedToken, declaration: snippet }).toMatchSnapshot();
    } finally {
      if (hadPrevious) {
        root.style.setProperty("--glitch-overlay-opacity-card", previous);
      } else {
        root.style.removeProperty("--glitch-overlay-opacity-card");
      }
    }
  });
});
