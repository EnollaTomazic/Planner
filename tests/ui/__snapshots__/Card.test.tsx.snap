// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Card > maps the glitch overlay opacity to the card token 1`] = `
{
  "computedToken": "",
  "declaration": ".glitch { --glitch-overlay-opacity: var(--glitch-overlay-opacity-card); }",
}
`;
